File: frontend/src/api/aiSkill.ts
import request from '@/utils/request'

export interface AISkill {
  id: number
  user_id: number
  name: string
  description: string
  generate_type: 'functional' | 'api'
  system_prompt: string
  user_prompt: string
  is_default: boolean
  created_at: string
  updated_at?: string
}

export interface AISkillCreate {
  name: string
  description?: string
  generate_type: 'functional' | 'api'
  system_prompt: string
  user_prompt: string
  is_default?: boolean
}

export interface AISkillUpdate {
  name?: string
  description?: string
  generate_type?: 'functional' | 'api'
  system_prompt?: string
  user_prompt?: string
  is_default?: boolean
}

// 获取技能列表
export const getAISkills = (generateType?: string) => {
  return request<AISkill[]>({
    url: '/ai-skills/',
    method: 'get',
    params: generateType ? { generate_type: generateType } : {}
  })
}

// 创建技能
export const createAISkill = (data: AISkillCreate) => {
  return request<AISkill>({
    url: '/ai-skills/',
    method: 'post',
    data
  })
}

// 更新技能
export const updateAISkill = (id: number, data: AISkillUpdate) => {
  return request<AISkill>({
    url: `/ai-skills/${id}`,
    method: 'put',
    data
  })
}

// 删除技能
export const deleteAISkill = (id: number) => {
  return request({
    url: `/ai-skills/${id}`,
    method: 'delete'
  })
}

// 初始化默认技能
export const initDefaultSkills = () => {
  return request({
    url: '/ai-skills/init-defaults',
    method: 'post'
  })
}

File: server/schemas/ai_skill.py
from typing import Optional
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class AISkillCreate(BaseModel):
    """创建技能请求"""
    name: str
    description: Optional[str] = ""
    generate_type: str  # functional / api
    system_prompt: str
    user_prompt: str  # 必须包含 {input_content}
    is_default: bool = False


class AISkillUpdate(BaseModel):
    """更新技能请求"""
    name: Optional[str] = None
    description: Optional[str] = None
    generate_type: Optional[str] = None
    system_prompt: Optional[str] = None
    user_prompt: Optional[str] = None
    is_default: Optional[bool] = None


class AISkillOut(BaseModel):
    """技能响应"""
    model_config = ConfigDict(from_attributes=True)

    id: int
    user_id: int
    name: str
    description: Optional[str] = ""
    generate_type: str
    system_prompt: str
    user_prompt: str
    is_default: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
